src/charm.py

```diff
@@ -214,8 +207,8 @@ def _process_outstanding_certificate_requests(self) -> None:
         """Process outstanding certificate requests."""
         for request in self.tls_certificates.get_outstanding_certificate_requests():
             self._generate_self_signed_certificate(
-                csr=request.csr,
-                is_ca=request.is_ca,
+                csr=str(request.certificate_signing_request),
```

Comment:
why not `request.certificate_signing_request.raw`?

Reply:
I mean, those two are equivalent, I just find that str(abc) looked cleaner than abc.raw. I don't really care either way.